sktime/forecasting/compose/_column_ensemble.py

```diff
@@ -0,0 +1,149 @@
+                        ("ses", ExponentialSmoothing(trend='add'), 1)]
+    >>> forecaster = ColumnEnsembleForecaster(forecasters=forecasters)
+    >>> forecaster.fit(y, fh=[1, 2, 3])
+    ColumnEnsembleForecaster(...)
+    >>> y_pred = forecaster.predict()
+    """
+
+    _required_parameters = ["forecasters"]
+    _tags = {
+        "scitype:y": "both",
+        "univariate-only": False,
+        "y_inner_mtype": "pd.DataFrame",
+        "requires-fh-in-fit": False,
+        "handles-missing-data": False,
+    }
+
+    def __init__(self, forecasters):
+        self.forecasters = forecasters
+        super(ColumnEnsembleForecaster, self).__init__(forecasters=forecasters)
+
+    @property
+    def _forecasters(self):
+        """Make internal list of forecasters.
+
+        The list only contains the name and forecasters, dropping
+        the columns. This is for the implementation of get_params
+        via _HeterogenousMetaEstimator._get_params which expects
+        lists of tuples of len 2.
+        """
+        return [(name, forecasters) for name, forecasters, _ in self.forecasters]
+
+    @_forecasters.setter
+    def _forecasters(self, value):
+        self.forecasters = [
+            (name, forecasters, columns)
+            for ((name, forecasters), (_, _, columns)) in zip(value, self.forecasters)
+        ]
+
+    def _fit(self, y, X=None, fh=None):
+        """Fit to training data.
+
+        Parameters
+        ----------
+        y : pd.DataFrame
+            Target time series to which to fit the forecaster.
+        fh : int, list or np.array, optional (default=None)
+            The forecasters horizon with the steps ahead to to predict.
+        X : pd.DataFrame, optional (default=None)
+            Exogenous variables are ignored.
+
+        Returns
+        -------
+        self : returns an instance of self.
+        """
+        names, forecasters, indices = self._check_forecasters()
+        if len(set(indices)) != len(indices):
+            raise ValueError(
+                "One estimator per column required. Found %s unique"
+                " estimators" % len(set(indices))
+            )
+        elif len(set(indices)) != len(y.columns):
+            raise ValueError(
+                "One estimator per column required. Found %s" % len(indices)
+            )
+        self.forecasters_ = []
+
+        for (name, forecaster, index) in self.forecasters:
+            forecaster_ = clone(forecaster)
+
+            forecaster_.fit(y.iloc[:, index], X, fh)
+            self.forecasters_.append((name, forecaster_, index))
+
+        return self
+
+    def _predict(self, fh=None, X=None, return_pred_int=False, alpha=DEFAULT_ALPHA):
+
+        y_pred = np.zeros((len(fh), len(self.forecasters_)))
+        for (_, forecaster, index) in self.forecasters_:
+            y_pred[:, index] = forecaster.predict(fh)
+
+        y_pred = pd.DataFrame(data=y_pred)
+        y_pred.index = self.fh.to_absolute(self.cutoff)
```

Comment:
We should probably also make sure that we return the same column names as seen in `fit`.